Source/WebKit/NetworkProcess/NetworkConnectionToWebProcess.cpp

```diff
@@ -958,13 +958,28 @@ void NetworkConnectionToWebProcess::cookieEnabledStateMayHaveChanged()
     protectedConnection()->send(Messages::NetworkProcessConnection::UpdateCachedCookiesEnabled(), 0);
 }
 
+bool NetworkConnectionToWebProcess::isFilePathAllowed(NetworkSession& session, String path)
+{
+    path = FileSystem::lexicallyNormal(path);
+    auto parentPath = FileSystem::parentPath(path);
+    while (parentPath != path) {
+        if (m_allowedFilePaths.contains(path) || parentPath == session.storageManager().path() || parentPath == session.storageManager().customIDBStoragePath())
+            return true;
+        path = parentPath;
+        parentPath = FileSystem::parentPath(path);
+    }
+    return false;
```

Comment:
I don't think this always works. When the path was allowed from WebProcessProxy::assumeReadAccessToBaseURL() the base path contains a trailing slash, but here FileSystem::parentPath() returns the parent directory without a trailing slash. Since we compare strings here, `/tmp/ != /tmp`